src/shogun/features/DenseFeatures.cpp

```diff
@@ -865,6 +865,33 @@ SGVector<ST> CDenseFeatures<ST>::dot(const SGVector<ST>& other) const
 	return linalg::matrix_prod(get_feature_matrix(), other, false);
 }
 
+template <typename ST>
+void CDenseFeatures<ST>::dot(
+	SGVector<float64_t> output, IntRange feat_range,
```

Comment:
I wonder: isn't "range" something that could be achieved with subsets? Instead of passing the parameter around?

Reply:
I thought about this. subsets are not ranges, we only provide functionality for subsets. To represent a range, we would have to create an array with every single index in the range, so treating ranges as subsets isn't the best way I think.

Reply:
It would be better imo if we provided functionality to "slice" a feature object using a range, instead of individual indices. We can start with DenseFeatures and other features where this clear how to do efficiently, and the other features make them reduce to the subset behaviour.

Reply:
yeah I agree the implementation would have to account for that. But it could right?
I am more talking about the conceptual interface. 
Subset is a class that could be overloaded btw ....

Reply:
alright, so a subset can either have an array of indices, or a range. if it's a range, it should be able to do the same functionality as the one with indices so not to break things.
we add a view function that accepts a range instead of SGVector of indices
we remove the IntRange thing, and also remove the other "dot" method that multiplies with the whole matrix, and leave this with the "alphas" and "bias"

Reply:
@karlnapf sounds reasonable?

Reply:
Yes polymorphic subsets/views
And the we get a cleaner api

I wonder whether there are downsides to this? You recently looked at all the dot range methods, something came up?
Only thing I can think of are the combined features where a view would shallow copy all sub features...